holmes/interactive.py

```
@@ -2251,6 +2251,28 @@ def _wait_for_completion_or_escape(
+def _run_hooks(hooks: Optional[dict], hook_name: str) -> None:
+    """Fire-and-forget execution of configured lifecycle hook commands."""
+    if not hooks:
+        return
+    for hook in hooks.get(hook_name, []):
+        try:
+            if not isinstance(hook, dict):
+                logging.warning("Skipping invalid %s hook (expected dict, got %r)", hook_name, hook)
+                continue
+            cmd = hook.get("command")
+            if cmd:
+                subprocess.Popen(
```

Ruff (0.15.12)

[error] `subprocess` call with `shell=True` identified, security issue

(S602)